diff --git a/src/services/machines.ts b/src/services/machines.ts
index 2585835a671c6582473e6d798daadc06af6bfad6..36b79819730f579dd7101fda9432bfe8b9486489 100644
--- a/src/services/machines.ts
+++ b/src/services/machines.ts
@@ -575,7 +575,7 @@ export const CertificateDetail = /*@__PURE__*/ S.suspend(() =>
     dns_provider: S.optional(S.String),
     dns_requirements: S.optional(DNSRequirements),
     hostname: S.optional(S.String),
-    rate_limited_until: S.optional(S.String),
+    rate_limited_until: S.optional(S.NullOr(S.String)),
     status: S.optional(S.String),
     validation: S.optional(CertificateValidation),
     validation_errors: S.optional(CertificateDetailValidationErrorsList),
@@ -673,10 +673,10 @@ export interface IPAssignment {
 }
 export const IPAssignment = /*@__PURE__*/ S.suspend(() =>
   S.Struct({
-    created_at: S.optional(S.String),
+    created_at: S.optional(S.NullOr(S.String)),
     ip: S.optional(S.String),
-    region: S.optional(S.String),
-    service_name: S.optional(S.String),
+    region: S.optional(S.NullOr(S.String)),
+    service_name: S.optional(S.NullOr(S.String)),
     shared: S.optional(S.Boolean),
     type: S.optional(S.String),
   }),
